<?xml version="1.0"?>
<robot name="Gen3" xmlns:xacro="http://ros.org/wiki/xacro">
  <webots>
    <plugin type="webots_ros2_control::Ros2Control"/>
  </webots>
  <ros2_control name="WebotsControl" type="system">
    <hardware>
      <plugin>webots_ros2_control::Ros2ControlSystem</plugin>
    </hardware>

    <!-- Gen3 -->
    <joint name="joint_1">
      <command_interface name="position">
        </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="joint_2">
      <command_interface name="position">
        </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="joint_3">
      <command_interface name="position">
        </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="joint_4">
      <command_interface name="position">
        </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="joint_5">
      <command_interface name="position">
        </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="joint_6">
      <command_interface name="position">
        </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>

    <!-- ROBOTIQ Gripper -->
    <joint name="left_finger_joint">
      <command_interface name="position">
        </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
    <joint name="right_finger_joint">
      <command_interface name="position">
        </command_interface>
      <state_interface name="position"/>
      <state_interface name="velocity"/>
    </joint>
  </ros2_control>
</robot>
